core/spec/models/spree/preferences/static_model_preferences_spec.rb

```diff
@@ -5,7 +5,7 @@
 module Spree
   RSpec.describe Preferences::StaticModelPreferences do
     let(:preference_class) do
-      Class.new do
+      MyFakeStaticModelPreferencesClass ||= Class.new do
```

Comment:
I think defining a constant here will actually define it under `Spree::`, an alternative could be using stubbed constants https://rspec.info/blog/2012/06/constant-stubbing-in-rspec-2-11/, but I'm not sure it's a good one, you need to check how they'll interact with this setup.